src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DecisionDagRewriter.cs

```diff
@@ -410,53 +412,511 @@ private void GenerateTest(BoundExpression test, BoundDecisionDagNode whenTrue, B
+                    }
+
+                    return (whenTrueBuilder.ToImmutableAndFree(), whenFalseBuilder.ToImmutableAndFree());
+                }
+            }
+
+            /// <summary>
+            /// A node in a tree representing the form of a generated decision tree for classifying an input value.
+            /// </summary>
+            private abstract class ValueDispatchNode
+            {
+                protected virtual int Height => 1;
+                public readonly SyntaxNode Syntax;
+
+                public ValueDispatchNode(SyntaxNode syntax) => Syntax = syntax;
+
+                /// <summary>
+                /// A node representing the dispatch by value (equality). This corresponds to a classical C switch
+                /// statement, except that it also handles values of type float, double, decimal, and string.
+                /// </summary>
+                internal sealed class SwitchDispatch : ValueDispatchNode
+                {
+                    public readonly ImmutableArray<(ConstantValue value, LabelSymbol label)> Cases;
+                    public readonly LabelSymbol Otherwise;
+                    public SwitchDispatch(SyntaxNode syntax, ImmutableArray<(ConstantValue value, LabelSymbol label)> dispatches, LabelSymbol otherwise) : base(syntax)
+                    {
+                        this.Cases = dispatches;
+                        this.Otherwise = otherwise;
+                    }
+                    public override string ToString() => "[" + string.Join(",", Cases.Select(c => c.value)) + "]";
+                }
+
+                /// <summary>
+                /// A node representing a final destination that requires no further dispatch.
+                /// </summary>
+                internal sealed class LeafDispatchNode : ValueDispatchNode
+                {
+                    public readonly LabelSymbol Label;
+                    public LeafDispatchNode(SyntaxNode syntax, LabelSymbol Label) : base(syntax) => this.Label = Label;
+                    public override string ToString() => "Leaf";
+                }
+
+                /// <summary>
+                /// A node representing a dispatch based on a relational test of the input value by some constant.
+                /// Nodes of this kind are required to be height-balanced when constructed, so that when the full
+                /// decision tree is produced it generates a balanced tree of comparisons.
+                /// </summary>
+                internal sealed class RelationalDispatch : ValueDispatchNode
+                {
+                    private int _height;
+                    protected override int Height => _height;
+                    public readonly ConstantValue Value;
+                    public readonly BinaryOperatorKind Operator;
+                    /// <summary>The side of the test handling lower values. The true side for &lt; and &lt;=, the false side for > and >=.</summary>
```

Comment:
we don't have to escape `>` in this context?

Reply:
As far as I can tell, no.  If there wasn't a preceding `<` to match it against, it is just treated as text.